### PR TITLE
refactor: add turborepo and upgrade to Node 16

### PR DESCRIPTION
**Pull request by andylangridge**
- Have added turbo repo in order to help clean up the npm scripts - https://turborepo.org/
- Also upgraded the project to Node 16 in the process
- Updated the `docker-compose.yml` file to use the actual Dockerfiles
- Moved the API and Web apps inside an "apps" folder, and kept "packages" for things like the common package.

### CODE REVIEW

**Comment by JamesP-KC** [comment]
1. reverted renaming of `packages` directory to `apps`, for ease of reviewing and keeping PR as smaller

2. fixed api 500 error

**Comment by mauriciokin** [comment]
e2e-tests/README.md - this file has make install and make serve references on line 36.

**Comment by mauriciokin** [comment]
When I run: npm run dev and do a CTRL + C after the server spins and runs I get this message:

```
^C
Gracefully stopping... (press Ctrl+C again to force)
Killing applications-web-app      ... done
Killing redis                     ... done
Killing applications-service-api  ... done
Killing ni_db                     ... done
Killing adminer                   ... done
ERROR: 2
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! applications-service@ dev: `docker-compose up`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the applications-service@ dev script.
```

**Comment by mauriciokin** [comment]
Many files on docker-compose.yaml are exposing secrets and endpoints. 

As it's a public repo and even private ones for security reasons use env variables to pass those secrets using a secret manager.

It's unrelated to this task as it was already done like this in case it cannot or would take much effort to be done here I am happy to create a new ticket for it in the backlog.

**Comment by JamesP-KC** [comment]
> e2e-tests/README.md - this file has make install and make serve references on line 36.

Thanks! I've updated to remove that reference.



> When I run: npm run dev and do a CTRL + C after the server spins and runs I get this message:
> 
> ```
> ^C
> Gracefully stopping... (press Ctrl+C again to force)
> Killing applications-web-app      ... done
> Killing redis                     ... done
> Killing applications-service-api  ... done
> Killing ni_db                     ... done
> Killing adminer                   ... done
> ERROR: 2
> npm ERR! code ELIFECYCLE
> npm ERR! errno 2
> npm ERR! applications-service@ dev: `docker-compose up`
> npm ERR! Exit status 2
> npm ERR! 
> npm ERR! Failed at the applications-service@ dev script.
> ```

I've had a look into this, and I think it is down to the exit code that `docker-compose` returns after you hit CTRL+C, and that `npm` treats that exit code as an error. Ideally, we wouldn't see an error after a graceful exit like this but without doing some hacking around with the docker-compose exit code I'm not sure there's an obvious solution to this. Or we can just run `docker-compose up` directly, the `npm run dev` task isn't doing a lot so perhaps isn't needed anyway.


> Many files on docker-compose.yaml are exposing secrets and endpoints.
> 
> As it's a public repo and even private ones for security reasons use env variables to pass those secrets using a secret manager.
> 
> It's unrelated to this task as it was already done like this in case it cannot or would take much effort to be done here I am happy to create a new ticket for it in the backlog.

Definitely agree. We have/had a ticket in the backlog to move the value for the `SESSION_KEY` environment variable into Azure Key Vault. We should do the same for other secret values, and raise a separate ticket for those.

**Review by christopher-semmens — DISMISSED** [review]

File: packages/forms-web-app/src/app.js [hunk 1]
```diff
@@ -84,12 +89,10 @@ env.addFilter('formatMimeType', fileTypeDisplayHelper);
 env.addFilter('filterByKey', filterByKey);
 env.addFilter('addKeyValuePair', addKeyValuePair);
 env.addFilter('render', renderTemplateFilter(nunjucks));
-
-env.addGlobal('featureFlag', config.featureFlag);
 env.addGlobal('googleAnalyticsId', config.server.googleAnalyticsId);
 env.addGlobal('googleTagManagerId', config.server.googleTagManagerId);
+env.addGlobal('featureFlag', config.featureFlag);
 env.addGlobal('host', config.server.host);
-env.addGlobal('nsipBaseUrl', config.server.nsipBaseUrl);
```
[the remark this answers is not in the document]
Comment (christopher-semmens) [reply packages/forms-web-app/src/app.js:92]: Hi @JamesP-KC /unit/middleware/decode-uri.test.js is also being deleted in this PR. That file needs to remain.

[the remark this answers is not in the document]
Comment (JamesP-KC) [reply packages/forms-web-app/src/app.js:92]: thanks @christopher-semmens , good spot. Some rebasing gone awry there... should be resolved now

**Review by christopher-semmens — APPROVED** [review]

**Pull request merged**